tests/run-engine-taurus-deploy.sh

```diff
@@ -0,0 +1,209 @@
+}
+
+# Parse args passed in by run-fvttests.sh script when calling this script with the -o and -t options. 
+# See usage statement above for details
+
+if [[ $# -lt 4 ]] # Requires at least -o <val> -t <val>
+then
+  usage
+fi
+while getopts "o:t:" arg; do 
+  case ${arg} in 
+    o)
+      echo "Final output ${OPTARG}"
+      outputXML=${OPTARG}
+      ;;
+    t)
+      echo "Test files ${OPTARG}"
+      testFiles=(`echo ${OPTARG} | tr ',' ' '`) # Convert into array
+      ;;
+    ?)
+      echo "Invalid option"
+      usage
+      exit -1
+      ;;
+  esac
+done
+
+# Clean up from previous runs -- Delete the pod that was spun off to run the tests on the prior run
+# Also remove the output directory which was created to hold the results locally accessible inside of the cd pipeline
+rm -rf ${OUTPUT_DIR}
+kubectl -n ${CLUSTER_NAMESPACE} delete pod/${POD_NAME}
+
+# Get the deployment file ready for the pod that will be spun off.
+cp run-engine-taurus-deploy.yaml run-engine-taurus-deploy-with-replaced-values.yaml
+sed -i "s/CLUSTER_NAMESPACE_PLACEHOLDER/${CLUSTER_NAMESPACE}/g" run-engine-taurus-deploy-with-replaced-values.yaml
+sed -i "s/POD_NAME_PLACEHOLDER/${POD_NAME}/g" run-engine-taurus-deploy-with-replaced-values.yaml
+sed -i "s/TRUSTSTORE_PLACEHOLDER/${TRUSTSTORE}/" run-engine-taurus-deploy-with-replaced-values.yaml
+sed -i "s/TRUSTSTORE_TYPE_PLACEHOLDER/${TRUSTSTORE_TYPE}/" run-engine-taurus-deploy-with-replaced-values.yaml
+
+cd ${TEST_DIR}
+
+# Get the cohort-cli shaded jar from the cohort-services pod.
+COHORTSVC_POD=$(kubectl get pods --namespace "${CLUSTER_NAMESPACE}" | grep -i "${APP_NAME}" | grep Running | cut -d " " -f 1 | head -n1)
+kubectl -n ${CLUSTER_NAMESPACE} cp ${COHORTSVC_POD}:/opt/alvearie/cohortEngine/cohort-cli-shaded.jar ${LIB_DIR}/cohort-cli-shaded.jar
+echo "Copied cohort-cli shaded jar from the running cohort-services pod."
+
+# Before tests can be executed prepare the test yaml files to have the right values replaced in them  
+
+cd ${CONFIG_DIR}
+# Generate fhir config file for test-fvt tenant
+cp local-ibm-fhir.json fhirconfig-testfvt-tenant.json
+sed -i '/\"password\"/s/:.*$/: \"'"${FHIR_USER_PASS}"'\",/' fhirconfig-testfvt-tenant.json
+sed -i '/\"endpoint\"/s/:.*$/: \"'"${FHIR_ENDPOINT}"'\",/' fhirconfig-testfvt-tenant.json
+
+#generate updated default tenant fhir config json
+cp local-ibm-fhir.json fhirconfig-default-tenant.json
+sed -i '/\"password\"/s/:.*$/: \"'"${FHIR_USER_PASS}"'\",/' fhirconfig-default-tenant.json
+sed -i '/\"tenantId\"/s/:.*$/: \"default\",/' fhirconfig-default-tenant.json
+sed -i '/\"endpoint\"/s/:.*$/: \"'"${FHIR_ENDPOINT}"'\",/' fhirconfig-default-tenant.json
+
+cd ${SCENARIOS_DIR}
+# update breastCancerTestScenarios yaml file for FHIR details, path to the cohort-cli shaded jar and the final xml file to be generated.
+sed -i '/JAR/s/:.*$/: '"${SHADED_JAR}"'/' breastCancerTestScenarios.yaml
```

Comment:
My immediate reaction to seeing these blocks of repeating replace operations was, "there has to be a better way." It turns out that there are several better ways to do search and replace in a single execution of sed.  The one I am partial to is something like sed -i -e '...' -e '...' -e '...' where each -e is the replace operation you want to perform. The upside to an approach like this is that it will make the script run faster.  In this particular situation, I expect the tests will take up the bulk of the run time, so nothing blocking. Just FYI for future.

Reply:
Thanks for suggestion